### PR TITLE
Re write the SampleBatchEntrySetup Page in React; created SampleBatch…

### PR DESCRIPTION
**Pull request by josephbate**
## Description of what I have done 

1. i created a rest controller `SampleBatchEntrySetupRestController.java`

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by josephbate** [comment]
hello @mozzy11 
please review this commit

**Comment by mozzy11** [comment]
@josephbate can you make your PR a WIP ??

**Comment by mozzy11** [comment]
Thanks @josephbate . 
Can you go ahead and add the UI and attach a screenshot ??

File: src/main/java/org/openelisglobal/samplebatchentry/controller/rest/SampleBatchEntrySetupRestController.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+    @Autowired
+    SiteInformationService siteInformationService;
+
+    @InitBinder
+    public void initBinder(WebDataBinder binder) {
+        binder.setAllowedFields(ALLOWED_FIELDS);
+    }
+
+    @RequestMapping(value = "ReactSampleBatchEntrySetup" ,produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (mherman22) [inline src/main/java/org/openelisglobal/samplebatchentry/controller/rest/SampleBatchEntrySetupRestController.java:39]:
```suggestion
    @RequestMapping(value = "ReactSampleBatchEntrySetup" method = RequestMethod.GET,produces = MediaType.APPLICATION_JSON_VALUE)
```

**Review by mherman22 — CHANGES_REQUESTED** [review]

File: src/main/java/org/openelisglobal/samplebatchentry/controller/rest/SampleBatchEntrySetupRestController.java (hunk 1 above)
Comment (josephbate) [reply src/main/java/org/openelisglobal/samplebatchentry/controller/rest/SampleBatchEntrySetupRestController.java:39]: thank you let me make cahnnges @mherman22

**Comment by josephbate** [comment]
hello @mozzy11 
this is my progress so far 
Screenshot from 2024-03-18 09-52-07

**Pull request closed without merging**